### DISCUSSION TITLE
Unit Commitment

### DISCUSSION THREAD

**Discussion by A-A-123** [yalmip/YALMIP#1347] [Problems using YALMIP] [answered]
Following a proposed tutorial on **unit commitment** (https://yalmip.github.io/example/unitcommitment/). I do have two questions please:

1. In case another factor (**Alpha**) needs to be included in the model, then how can this happen? Currently, cost function is (C*P+Q*P^2), and it is proposed to have it as (**Alpha**+C*P+Q*P^2), where Alpha is simply fixed and does not depend on P but it is deafferent for each generator unit. Should the objective be as:
Objective = Objective +**Alpha*onoff(:,k)**+ P(:,k)'*Q*P(:,k) + C*P(:,k);

2. I keep getting different results by changing the initial value of Objective from **Objective=0** to **Objective=[]**, and the stairs(value(P)') is totally different for each case, why is that (I have another case where the code shows error if Objective=0 but it runs fine when Objective=[])?
When Objective=[], the following message is shown but no list of iterations:
"_ptimal solution found. Intlinprog stopped at the root node because the objective value is within a gap tolerance of the optimal value, options.AbsoluteGapTolerance = 0. The intcon variables are integer within tolerance, options.IntegerTolerance = 1e-05._"

**Comment by johanlofberg** [comment, marked as answer]
1. Yes
2. Initializing to [] makes no sense because you cannot add to an empty array

```

>> []+1

ans =

     []

```
